docs/tools/CLI/build_profiles.md

```diff
@@ -1,42 +1,42 @@
 <h2 id="build-profiles">Build profiles</h2>
 
-Arm Mbed OS 5 supports three primary build profiles: *develop*, *debug* and *release*. The Online Compiler uses the *develop* profile. When building from Arm Mbed CLI, you can select a profile by adding the `--profile <profile>` flag. You can specify custom user-defined profiles by giving the path to the profile.
+Arm Mbed OS 5 defines three collections of toolchain flags used during the build. These are __build profiles__. These are collections of toolchain flags used during the build. The three build profiles are *develop*, *debug* and *release*. The Mbed Online Compiler uses the *develop* build profile. When building from Arm Mbed CLI, you can select the build configuration by adding the `--profile <`__build profile__ `name>` flag. You can also specify custom- or user-defined configurations by giving the path to the JSON file defining the build configuration.
 
-### Develop profile
+### Develop
 
 - Small and fast code.
 - Full error information. For example, asserts have file name and line number.
 - Hard to follow code flow when using a debugger.
 - Chip goes to sleep when idle:
-    * Debugger is likely to drop connection.
-    * Breaks the local file system on the [Arm Mbed interface](../introduction/index.html) on some boards.
+   - Debugger is likely to drop connection.
+   - Breaks the local file system on the [Arm Mbed interface](../introduction/index.html) on some boards.
 
-### Debug profile
+### Debug
 
-- Largest and slowest profile.
+- Largest and slowest performance.
 - Full error information. For example, asserts have file name and line number.
 - Easy to step through code with a debugger.
 - Disabled sleep mode.
 
-### Release profile
+### Release
 
-- Smallest profile and still fast.
+- Smallest code size and still fast.
 - Minimal error information.
 - Chip goes to sleep when going idle:
-    - Debugger is likely to drop connection.
-    - Breaks the local file system on the [Mbed interface](../introduction/index.html) on some boards.
+   - Debugger is likely to drop connection.
+   - Breaks the local file system on the [Mbed interface](../introduction/index.html) on some boards.
 
-### User defined profiles
+### User-defined build profile
 
-A toolchain or build system profile is a set of flags that is guaranteed to be passed to the underlying compiler suite.
+As mentioned above, the __build profile__ defines the set of flags that is guaranteed to be passed to the underlying compiler suite.
 
 These flags are stored in a JSON file that may be merged with other JSON files of the same structure.
 
-#### JSON toolchain profile format
+#### JSON build profile format
 
-The JSON object that represents a toolchain profile is a dictionary mapping from toolchains, such as `GCC_ARM`, to their flags, such as `-O3`.
+The JSON object that represents toolchain configurations for mapping each supported toolchains, such as `GCC_ARM`, to their flags, like `-O3`.
```

Comment:
I think this sentence lost a subject. Also, "like" should probably be "such as", unless it can't be -03. ("Like" means similar to but not including, and "such as" means similar to and including.)

Reply:
@AnotherButler - I made some changes to clarify the questions, please review again.